### PR TITLE
KAFKA-7266: Fix MetricsTest.testMetrics flakiness by increasing batch…

### PR DESCRIPTION
**Pull request by stanislavkozlovski**
The test `kafka.api.MetricsTest.testMetrics` has been failing intermittently in kafka builds.
The particular failure is in the `MessageConversionsTimeMs` metric assertion:
```
java.lang.AssertionError: Message conversion time not recorded 0.0
```

The test would intermittently fail because the conversion time took less than 1 millisecond. Such conversion time is reported as zero by the metric.
There has been work done previously (https://github.com/apache/kafka/pull/4681) to combat the flakiness of the test and while it has improved it, the test still fails sometimes.

Locally, this test failed 5 times out of 25.

### Changes
Increase the record size and use compression - both will slow down message conversion enough to have it be above 1ms. Locally this test has not failed in 200 runs and counting

### CODE REVIEW

**Comment by stanislavkozlovski** [comment]
cc @ijuma @rajinisivaram for review please

File: core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala
Comment (rajinisivaram) [inline core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala]: Rather than add a new field and treat as a special case, we could just get the value from `producerConfig` and use that in `createProducer`?

File: core/src/test/scala/integration/kafka/api/MetricsTest.scala
Comment (rajinisivaram) [inline core/src/test/scala/integration/kafka/api/MetricsTest.scala]: Do we need this check at all? We will just be sending more batches without the check and that should be ok?

**Review by rajinisivaram — COMMENTED** [review]
@stanislavkozlovski Thanks for the PR. Looks good, left a couple of minor comments.

**Comment by ijuma** [comment]
Maybe set compression type to gzip to make conversion slower. That may be all that is needed.

File: core/src/test/scala/integration/kafka/api/MetricsTest.scala
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/MetricsTest.scala]: A way to ensure your batch is large is to use large records. Then you don't need this at all.

Comment (stanislavkozlovski) [reply core/src/test/scala/integration/kafka/api/MetricsTest.scala]: True, I wrongly thought this was the minimum batch size. Should have read the documentation

**Review by ijuma — APPROVED** [review]
LGTM if the test is now stable. Will leave it to @rajinisivaram to merge.

**Comment by rajinisivaram** [comment]
retest this please

**Review by rajinisivaram — APPROVED** [review]
@stanislavkozlovski Thanks for the PR, LGTM. Merging to trunk.

**Pull request merged**
